### DISCUSSION TITLE
How can I do regex validation to field value

### DISCUSSION THREAD

**Discussion by t-kawatsu** [directus/directus#8574] [Archive]
I want to ensure a field value to /[A-Z]+/ format.
For doing that, I want to set regex validation on the field.
But I do not know how to do that on V9 version.
How can I do that?

Thanks.

**Comment by joselcvarela** [comment]
Hello @t-kawatsu,
I think the validation of a fields is only possible via custom hooks.
So here is an example:
```js
// extensions/hooks/regex-validation/index.js

import { FailedValidationException } from 'directus';

function beforeSave(method) {
  return function (input, { collection, schema }) {
    if (collection !== 'collection_with_regex_field') return input;

    if (!/[A-Z+]/.test(input.regex_field))
      throw new FailedValidationException({
        message: 'Regex field has not a valid format',
        path: ['regex_field'],
        type: 'custom.pattern.base',
      });

    return input;
  };
}

module.exports = function registerHook({ services, exceptions }) {
  return {
    'items.create.before': beforeSave('create'),
    'items.update.before': beforeSave('update'),
  };
};
```
cc @alesvaupotic

**Reply by t-kawatsu** [reply]
thank you

**Reply by benhaynes** [reply]
Adding here that we already have _Field Validation_ that supports RegEx within a Role's custom permission settings.

**Reply by t-kawatsu** [reply]
I  see. thank you

**Reply by t-kawatsu** [reply]
@benhaynes
I could set validations to a collection field with Role’s custom permission.
But I felt It is little bit difficult to recognize that roles permissions can set collections field validations.

I think the permission settings can be set and own by the field or the collection itself. 
It is great to be able to set validation rules per roles, but why do you do so ?

thank you.

**Reply by benhaynes** [reply]
We will also be adding collection level and field level validation, but roll level came first. This is an ongoing process.
